Assets/MixedRealityToolkit.Tests/PlayModeTests/FocusProviderTests.cs

```diff
@@ -70,6 +70,29 @@ public IEnumerator TestGazeCursorArticulated()
             yield return null;
             Assert.IsTrue(inputSystem.GazeProvider.GazePointer.IsInteractionEnabled, "Gaze cursor should be visible after select command");
         }
+
+        /// <summary>
+        /// Ensure that the gaze provider hit result is not null when looking at an object,
+        /// even when the hand is up
+        /// </summary>
+        /// <returns></returns>
+        [UnityTest]
+        public IEnumerator TestGazeProviderTargetNotNull()
+        {
+            TestUtilities.PlayspaceToOriginLookingForward();
+            var cube = GameObject.CreatePrimitive(PrimitiveType.Cube);
+            cube.transform.position = Vector3.forward;
+
+            yield return null;
+
+            Assert.NotNull(MixedRealityToolkit.InputSystem.GazeProvider.GazeTarget, "GazeProvider target is null when looking at an object");
+
+            TestHand h = new TestHand(Handedness.Right);
+            yield return h.Show(Vector3.forward * 0.2f);
+            yield return null;
+
+            Assert.NotNull(MixedRealityToolkit.InputSystem.GazeProvider.GazeTarget, "GazeProvider target is null when looking at an object with hand raised");
```

Comment:
Is there also any way to validate that the the object isn't focused with the hand raised (i.e. that interaction enabled is properly being honored?)

Reply:
Yes, this is covered in [PointerEventTests::Test(). ](https://github.com/microsoft/MixedRealityToolkit-Unity/blob/f7f9e051327bc145012fc64804681f03d2cd709a/Assets/MixedRealityToolkit.Tests/PlayModeTests/PointerEventsTests.cs#L174)

```
            // Show hand far enough from the test collider to not give it focus
            var rightHand = new TestHand(Handedness.Right);
            Vector3 rightNoFocusPos = new Vector3(0.05f, 0, 1.0f);
            yield return rightHand.Show(rightNoFocusPos);

            // Should receive focus left events from gaze pointer
            expected.numBeforeFocusChange++;
            expected.numFocusChanged++;
            expected.numFocusExit++;
            handler.state.AssertEqual(expected);
```